api/client.go

```diff
@@ -107,11 +108,9 @@ func request[Req, Res any](client Client, method string, path string, query Quer
 
 	resp, err := client.HTTP.Do(req)
 	if err != nil {
-		urlErr := &url.Error{}
-		if errors.As(err, &urlErr) {
-			if urlErr.Timeout() {
-				return nil, fmt.Errorf("%w: %s", ErrTimeout, err)
-			}
+		connError := HandleConnError(err)
+		if connError != nil {
```

Comment:
very minor style comment. Generally we try to keep these in one line when we can:

```suggestion
		if connError := HandleConnError(err); connError != nil {
```